src/components/network/DockerNetworkModal.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import { useAsyncCallback } from 'react-async-hook';
+import { Form, Modal } from 'antd';
+import { usePrefixedTranslation } from 'hooks';
+import { useStoreActions, useStoreState } from 'store';
+import { Network } from 'types';
+import DockerNetworkName from 'components/common/DockerNetworkName';
+
+interface Props {
+  network: Network;
+}
+
+const DockerNetworkModal: React.FC<Props> = ({ network }) => {
+  const { l } = usePrefixedTranslation('cmps.network.actions.DockerNetworkModal');
+  const [form] = Form.useForm();
+  const { visible } = useStoreState(s => s.modals.dockerNetwork);
+  const { hideDockerNetwork } = useStoreActions(s => s.modals);
+
+  const { setDockerExternalNetworkName, getExternalDockerNetworks } = useStoreActions(
+    s => s.network,
+  );
+
+  const { notify } = useStoreActions(s => s.app);
+  const [isDockerNetworkNameValid, setIsDockerNetworkNameValid] = useState<boolean>(true);
+
+  const setExternalDockerNetworkAsync = useAsyncCallback(
+    async (id: number, externalNetworkName: string | undefined) => {
```

Comment:
The strings in this function should be moved into the `en-US.json` file.

Also, since the notification is displayed after the action has been taken, the messages should be in the past tense. Ex: Cleared, Created, Attached